### DISCUSSION TITLE
Using trait on non-user model - GuardDoesNotMatch Exception

### DISCUSSION THREAD

**Discussion by ShareghYusefi** [spatie/laravel-permission#2238] [Q&A]
I have a model that needs access to roles and found [#310](https://github.com/spatie/laravel-permission/issues/310) as a potential way of implementing this feature. However, I'm receiving the following exception: 

Spatie\Permission\Exceptions\GuardDoesNotMatch
`The given role or permission should use guard `` instead of ``.`

Is this workaround no longer being supported?

**Comment by systemsolutionweb** [comment]
Set a default guard_name
https://github.com/spatie/laravel-permission/pull/2203
